src/app/models/course.ts

```diff
@@ -0,0 +1,23 @@
+export interface Course {
+  id: string;
+  title: string;
+  createDate: Date;
+  duration: number;
+  description: string;
+}
+
+export class CourseMock implements Course {
+  readonly id!: string;
+  title!: string;
+  createDate!: Date;
+  duration!: number;
+  description!: string;
+
+  constructor(title: string, description: string, duration: number) {
+    this.id = Math.random().toString(20).substr(2, 5);
+    this.title = title;
+    this.description = description;
+    this.duration = duration;
+    this.createDate = new Date();
+  }
+}
```

Comment:
As I see there is no sense to describe `class` like this, you can just implement it like: 
```
export class CourseMock implements Course {
  readonly id!: string;
  createDate!: Date;

  constructor(
    public title: string,
    public description: string, 
    public duration: number
  ) {
    this.id = Math.random().toString(20).substr(2, 5);
    this.createDate = new Date();
  }
}
```
When you add a modifier inside constructor arguments, it automatically assigns them under the hood, for example: `constructor(public title: string) {}` it is the same as `constructor(title: string) {this.title = title}`

Reply:
If you want to use the `!` Operator for non-null checks, you can leave your own option, but I think that in this case there is no need

Reply:
Good to know. Thanks.